staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go

```diff
@@ -63,14 +63,16 @@ var _ discovery.CachedDiscoveryInterface = &CachedDiscoveryClient{}
 // ServerResourcesForGroupVersion returns the supported resources for a group and version.
 func (d *CachedDiscoveryClient) ServerResourcesForGroupVersion(groupVersion string) (*metav1.APIResourceList, error) {
 	filename := filepath.Join(d.cacheDirectory, groupVersion, "serverresources.json")
-	cachedBytes, err := d.getCachedFile(filename)
+	cachedBytes, cacheErr := d.getCachedFile(filename)
 	// don't fail on errors, we either don't have a file or won't be able to run the cached check. Either way we can fallback.
-	if err == nil {
+	if cacheErr == nil {
 		cachedResources := &metav1.APIResourceList{}
 		if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), cachedBytes, cachedResources); err == nil {
 			klog.V(10).Infof("returning cached discovery info from %v", filename)
 			return cachedResources, nil
 		}
+	} else {
+		klog.V(6).Infof("failed to read cache to %v due to %v", filename, cacheErr)
```

Comment:
> failed to read cache to %v due to

"failed to xxxx" is too strong a word for an info log.. i suggest sth like `cache file can be stale: %v`

and @caesarxuchao has the call to unhold

Reply:
Not just stale error, maybe it can be wrong file error. This statement is the same pattern with https://github.com/kubernetes/kubernetes/blob/ed8a260fb3ff03843e3e29bc8e0f24fe4bfb7ab4/staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go#L130

Reply:
How about
"failed to read cache **from** %v due to %v, **fallback to live discover**"